futures-util/src/stream/mod.rs

```diff
@@ -128,179 +128,198 @@ impl<T: ?Sized> StreamExt for T where T: Stream {}
 /// An extension trait for `Stream`s that provides a variety of convenient
 /// combinator functions.
 pub trait StreamExt: Stream {
-    /// Creates a future that resolves to the next element in the stream.
+    /// Creates a future that resolves to the next item in the stream.
     ///
     /// Note that because `next` doesn't take ownership over the stream,
-    /// the `Stream` type must be `Unpin`. If you want to use `next` with a
-    /// `!Unpin` stream, you'll first have to pin the stream. This can be
-    /// done by wrapping the stream in a `PinBox` or pinning it to the stack
-    /// using the `pin_mut!` macro.
+    /// the [`Stream`] type must be [`Unpin`]. If you want to use `next` with a
+    /// [`!Unpin`](Unpin) stream, you'll first have to pin the stream. This can
+    /// be done by wrapping the stream in a [`PinBox`](std::boxed::PinBox) or
+    /// pinning it to the stack using the `pin_mut!` macro.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #![feature(pin)]
+    /// #[macro_use] extern crate futures;
+    /// use futures::executor::block_on;
+    /// use futures::prelude::*;
+    /// use futures::stream;
+    ///
+    /// let stream = stream::iter(1..=3);
+    /// pin_mut!(stream);
+    ///
+    /// assert_eq!(block_on(stream.next()), Some(1));
+    /// assert_eq!(block_on(stream.next()), Some(2));
+    /// assert_eq!(block_on(stream.next()), Some(3));
+    /// assert_eq!(block_on(stream.next()), None);
```

Comment:
I added quite a few doc examples

Reply:
FYI there is `futures::executor::block_on_stream` which will turn a stream into an iterator.

Reply:
But not when I want to demonstrate the next combinator :)